js/app.js

```diff
@@ -40,87 +28,233 @@ var dogDuck = 0;
   } else if (answer === [4]) {
-    console.log(boots += 1);
+    boots++;
   } else if (answer === [5]) {
-    console.log(breakfast += 1);
+    breakfast++;
   } else if (answer === [6]) {
-    console.log(bubblegum += 1);
+    bubblegum++;
   } else if (answer === [7]) {
-    console.log(chair += 1);
+    chair++;
   } else if (answer === [8]) {
-    console.log(cthulhu += 1);
+    cthulhu++;
   } else if (answer === [9]) {
-    console.log(dogDuck += 1);
+    dogDuck++;
   } else if (answer === [10]) {
-    console.log(dragon += 1);
+    dragon++;
   } else if (answer === [11]) {
-    console.log(pen += 1);
+    pen++;
   } else if (answer === [12]) {
-    console.log(petSweep += 1);
+    petSweep++;
   } else if (answer === [13]) {
-    console.log(scissors += 1);
+    scissors++;
   } else if (answer === [14]) {
-    console.log(shark += 1);
+    shark++;
   } else if (answer === [15]) {
-    console.log(tauntaun += 1);
+    tauntaun++;
   } else if (answer === [16]) {
-    console.log(unicorn += 1);
+    unicorn++;
   } else if (answer === [17]) {
-    console.log(usb += 1);
+    usb++;
   } else if (answer === [18]) {
-    console.log(waterCan += 1);
+    waterCan++;
   } else if (answer === [19]) {
-    console.log(wineGlass += 1);
+    wineGlass++;
   } else {
-      alert('please click on image')
-  };
+    alert('please click on image');
+  }
+
+  if (attempts === maxAttempts) {
+    chart();
+  }
+});
+
+function ProductName(name, click, shown) {
+  this.name = name,
+  this.click = 0,
+  this.shown = 0
+}
 
 function generateRandomProduct() {
   var index = Math.floor(Math.random() * productNames.length);
   return productNames[index];
 }
+// function updateProductName (productName) {
+//   productNameElement.textcontent = productName;
+// }
+// function renderProductName (productName) {
+//   var h3 = document.createElement('h3');
+//   h3.textContent = productName;
+//   productNameParent.append('h3');
+// }
+//modeled from jon
+function buttonLogic() {
+  productName = [], answer = [];
+  for (var i = 0; i < productNames.length; i++){
+    productName.push(productNames[i].name);
+    answer.push(productNames[i]).clicked;
+  }
+}
+//return logic
+function data() {
+  // document.getElementById('data');
+  for (var i = 0; i < productNames.length; i++){
+    productName.push(productNames[i].name);
+    answer.push(productNames[i]).clicked;
+  }
+}
+
+for(var i = 0; i < localStorage.length; i++){
+  console.log(localStorage.key(i));
+}
+localStorage.setItem('clicked', 'data');
+localStorage.key();
+
+for (var n = 0; n < productNames.length; n++) {
+  arrayofObjects.push(ProductName = productNames[n]);
+}
+console.log(ProductName);
 
-function renderProductName (productName) {
-  var r = document.createElement('r');
-  r.textContent = productName;
-  productNameParent.append('r');
+// for (var i = 0; i < 3; i++) {
+//   arrayofObjects.push(productImage + i = ProductName);
+// }
+function updateProductName (productName) {
```

Comment:
Your constructors name is ProductName. It would be better to not name parameters or other things the same name, but just not capitalized. It makes the code a bit confusing. You also have two arrays named similar or the same: productName and productNames. I would challenge you to edit the names so parameters are not named the same as arrays or objects!